sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -990,27 +1008,23 @@ ur_program_handle_t ProgramManager::getBuiltURProgram(
     // emplace all subsets of the current set of devices into the cache.
     // Set of all devices is not included in the loop as it was already added
     // into the cache.
-    for (int Mask = 1; Mask < (1 << URDevicesSet.size()) - 1; ++Mask) {
```

Comment:
This code is much more maintainable than the new one. Also, if the problem is the "overflow" here, then it's inherent part of the problem. We'll have that many entries in the resulting set anyway (and likely OOM regardless).

Reply:
discussed in person, fixed in https://github.com/intel/llvm/pull/16334/commits/19983fe9bd65e3f9008cd3e244886c89dd547a3d